webrecorder/browsertrix
# Frontend workflow crawls list redesign

Partially addresses #720.



### Changes

- New list styles for Crawls and Workflows
- Hide running crawls in Crawls page
- Sort Workflows by Last Updated or Name
- Show last or current crawl state in Workflow
- Show name person who started Workflow
- Start, stop and cancel crawl from Workflow list

The minimum amount of changes needed to rework Crawl controls into Workflows have been implemented to reduce the scope of this PR.

### Manual testing

1. Run app and log in
2. Go to Crawls. Run a crawl. Verify running crawl is not shown in list.
3. Go to Workflows. Verify running Workflow is shown and duration is correct
4. Click running Workflow menu -> Stop Crawl. Verify crawl stops
5. Start another Workflow. Click running Workflow menu -> Cancel. Verify crawl cancels
6. Regression test: Duplicate Workflow, Copy Tags, Delete/Deactivate Workflow

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Workflows - List | <img width="1109" alt="Screen Shot 2023-04-17 at 2 54 50 PM" src="https://user-images.githubusercontent.com/4672952/232631256-ed1046da-59cc-4af3-b311-2a9a877aca3e.png"> |
| Workflows - Menu | <img width="259" alt="Screen Shot 2023-04-17 at 4 02 55 PM" src="https://user-images.githubusercontent.com/4672952/232631382-e8ed5cba-e312-4b67-86a3-2f23a079dec7.png"> |
| Workflows - Running Menu | <img width="264" alt="Screen Shot 2023-04-17 at 4 02 50 PM" src="https://user-images.githubusercontent.com/4672952/232631417-f03fbaac-ef86-437d-be2c-f79c7fd3fd47.png"> |
| Crawls - List | <img width="1143" alt="Screen Shot 2023-04-17 at 4 02 41 PM" src="https://user-images.githubusercontent.com/4672952/232631283-0d12e4ee-38ee-406c-908f-4c98e3b9307f.png"> |
| Crawls - Menu | <img width="230" alt="Screen Shot 2023-04-17 at 4 03 15 PM" src="https://user-images.githubusercontent.com/4672952/232631463-1eaa8dbb-a9cb-4881-ab49-6acd270c8862.png"> |

### Known issues
Besides the rest of the changes in #720, smaller screen sizes (<960px) aren't handled well and will need follow-up work to match tablet & mobile mockups.
